src/gh_range_diff.rs

```diff
@@ -0,0 +1,313 @@
+use std::collections::HashSet;
+use std::fmt::{self, Write};
+use std::sync::Arc;
+
+use anyhow::Context as _;
+use axum::{
+    extract::{Path, State},
+    http::HeaderValue,
+    response::IntoResponse,
+};
+use hyper::header::CACHE_CONTROL;
+use hyper::{
+    HeaderMap, StatusCode,
+    header::{CONTENT_SECURITY_POLICY, CONTENT_TYPE},
+};
+use imara_diff::{
+    Algorithm, Diff, InternedInput, Interner, Token, UnifiedDiffConfig, UnifiedDiffPrinter,
+};
+use pulldown_cmark_escape::FmtWriter;
+
+use crate::{github, handlers::Context, utils::AppError};
+
+pub async fn gh_range_diff(
```

Comment:
Could you please add some comment here to explain what it does, and explain what exactly should `basehead` contain?